### PR TITLE
Introduce pipeline documentation

### PR DESCRIPTION
**Pull request by maiste**
This PR introduces a diagram documentation about the pipeline and, how the steps interact together.

### CODE REVIEW

**Comment by maiste** [comment]
@TheLortex do you think we can merge it?

File: doc/pipeline-diagram.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+### Compute the set of dependencies
+
+```mermaid 
+stateDiagram-v2
+    solver: Solver
+    note left of solver
+        For each tuple (package, version), it computes
+        the dependencies.
+        Two solver steps:
+        - with ocaml-base-compiler >= 4.02.3
```
Comment (TheLortex) [inline doc/pipeline-diagram.md:150]: Currently there's also an upper bound on ocaml 5, as the preparation step currently doesn't work

File: doc/pipeline-diagram.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+- [ocaml-docs-ci GitHub](https://github.com/ocurrent/ocaml-docs-ci)
+- [Voodoo Github](https://github.com/ocaml-doc/voodoo)
+
+## Abstract
+
+First, the purpose of this `ocurrent` pipeline is to compile the documentation of every package in the "opamverse". To do so, it generates what we call a dependency universe. For each package (alongside with the version), the documentation is related to the packages whom our package has been compiled with. For each package, we have to compute at least a dependency universe in the form of a hash of all the (package, version) used to build it.
+
+Once we have compute our universes, we select the wider one and we use `odoc` to generate the `html` documentation and push it to [docs-data.ocaml.org](docs-data.ocaml.org). For example, for `irmin.2.7.2`, the documentation is stored [here](https://docs-data.ocaml.org/live/p/irmin/2.7.2/doc/Irmin/index.html). This `html` is then used by [ocaml.org](ocaml.org) to display the documentation to the user in the package explorer system.
+
+The following schema are going throught the build step in a more detailled way.
```
Comment (TheLortex) [inline doc/pipeline-diagram.md:16]: I believe there is no t at the end of through

File: doc/pipeline-diagram.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+## Links:
+
+- [ocaml-docs-ci GitHub](https://github.com/ocurrent/ocaml-docs-ci)
+- [Voodoo Github](https://github.com/ocaml-doc/voodoo)
+
+## Abstract
+
+First, the purpose of this `ocurrent` pipeline is to compile the documentation of every package in the "opamverse". To do so, it generates what we call a dependency universe. For each package (alongside with the version), the documentation is related to the packages whom our package has been compiled with. For each package, we have to compute at least a dependency universe in the form of a hash of all the (package, version) used to build it.
+
+Once we have compute our universes, we select the wider one and we use `odoc` to generate the `html` documentation and push it to [docs-data.ocaml.org](docs-data.ocaml.org). For example, for `irmin.2.7.2`, the documentation is stored [here](https://docs-data.ocaml.org/live/p/irmin/2.7.2/doc/Irmin/index.html). This `html` is then used by [ocaml.org](ocaml.org) to display the documentation to the user in the package explorer system.
```
Comment (TheLortex) [inline doc/pipeline-diagram.md:14]: computed

Comment (TheLortex) [reply doc/pipeline-diagram.md:14]: we select the one with the largest number of dependencies

**Review by TheLortex — APPROVED** [review]
Thanks for documenting the project! I have added minor comments.

**Comment by maiste** [comment]
Thanks for the review! I'll fix the typos and I'll merge :)

**Pull request merged**